storage/src/snapshot.rs

```diff
@@ -309,6 +314,16 @@ impl StateRead for Snapshot {
 /// A reader interface for rocksdb. NOTE: it is up to the caller to ensure consistency between the
 /// rocksdb::DB handle and any write batches that may be applied through the writer interface.
 impl TreeReader for Inner {
+    /// Gets a value by identifier, returning the newest value whose version is *less than or
+    /// equal to* the specified version.  Returns None if the value does not exist.
+    fn get_value_option(
```

Comment:
note: this probably should be a default implementation

Reply:
I don't think it can be a default implementation, the point of the `jmt` change was to have implementors store segmented data...

Reply:
perhaps something like:

```rust
pub struct KeyHash<H: SimpleHasher>(pub [u8; 32], PhantomData<H>);
```

?